File: tools/test_connections.js
const axios = require('axios');

async function testConnection(name, url) {
    console.log(`Testing ${name} (${url})...`);
    try {
        const response = await axios.get(url, {
            headers: {
                'User-Agent': 'AntigravityAINewsDashboard/1.0.0 (by /u/NoelS)'
            },
            timeout: 10000
        });
        if (response.status === 200) {
            console.log(`✅ ${name} responded successfully.`);
            return true;
        } else {
            console.log(`❌ ${name} returned status code ${response.status}.`);
            return false;
        }
    } catch (error) {
        console.log(`❌ ${name} failed: ${error.message}`);
        return false;
    }
}

async function run() {
    const sources = [
        ["Reddit", "https://www.reddit.com/r/ArtificialIntelligence/top.json?t=day"],
        ["Ben's Bites", "https://www.bensbites.com/archive"],
        ["AI Rundown", "https://www.therundown.ai/archive"]
    ];

    let allPassed = true;
    for (const [name, url] of sources) {
        const result = await testConnection(name, url);
        if (!result) allPassed = false;
    }

    if (allPassed) {
        console.log("\nAll links verified. Ready for architecture phase.");
        process.exit(0);
    } else {
        console.log("\nSome links failed. Check findings.md for details.");
        process.exit(1);
    }
}

run();

File: tools/test_connections.py
import requests
import sys

def test_connection(name, url):
    print(f"Testing {name} ({url})...")
    try:
        headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'}
        response = requests.get(url, headers=headers, timeout=10)
        if response.status_code == 200:
            print(f"✅ {name} responded successfully.")
            return True
        else:
            print(f"❌ {name} returned status code {response.status_code}.")
            return False
    except Exception as e:
        print(f"❌ {name} failed: {str(e)}")
        return False

if __name__ == "__main__":
    sources = [
        ("Reddit", "https://www.reddit.com/r/ArtificialIntelligence/top.json?t=day"),
        ("Ben's Bites", "https://www.bensbites.com/archive"),
        ("AI Rundown", "https://www.therundown.ai/archive")
    ]
    
    results = [test_connection(name, url) for name, url in sources]
    
    if all(results):
        print("\nAll links verified. Ready for architecture phase.")
        sys.exit(0)
    else:
        print("\nSome links failed. Check findings.md for details.")
        sys.exit(1)
